### PR TITLE
Use same thread for AsyncResultCallback

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Review by kokosing — COMMENTED** [review]
What is story (motivation) behind this change?

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java [hunk 1]
```diff
@@ -51,18 +52,14 @@ public Stats get()
...
+                return callback.awaitResult();
+            }
+        };
 
-            return lastStats.getAndUpdate(previousStats -> toStats((Statistics) executor.get(callback::awaitResult), previousStats));
-        }
-        catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-        catch (RuntimeException e) {
```
Comment (losipiuk) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:62]: error handling no longer needed?

**Comment by wendigo** [comment]
> What is the story (motivation) behind this change?

Investigation with @losipiuk and @findepi (could be related to https://github.com/square/okio/issues/624)

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java (hunk 1 above)
Comment (wendigo) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:62]: it's handled by Failsafe now (CheckedSupplier)

**Pull request closed without merging**